db/00000000000007_task_aggregation_counters.up.sql

```diff
@@ -0,0 +1,14 @@
+-- Per-task report aggregation counters, used for metrics.
+--
+-- Fillfactor is lowered to improve the likelihood of heap-only tuple optimizations. See the
+-- discussion around this setting for the task_upload_counters table.
+CREATE TABLE task_aggregation_counters(
+    id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,  -- artificial ID, internal only
+    task_id BIGINT NOT NULL,  -- task ID the counter is associated with
+    ord BIGINT NOT NULL,  -- the ordinal index of the task aggregation counter
+
+    success BIGINT NOT NULL DEFAULT 0,  -- reports successfully aggregated
+
+    CONSTRAINT task_aggregation_counters_unique_id UNIQUE(task_id, ord),
+    CONSTRAINT fk_task_id FOREIGN KEY(task_id) REFERENCES tasks(id) ON DELETE CASCADE
+) WITH (fillfactor = 50);
```

Comment:
Given that uploading these counter tables can be performance sensitive, I'm surprised we don't create an index on task ID. Maybe it's irrelevant because this table will be smallish anyway?

Reply:
`CONSTRAINT task_aggregation_counters_unique_id UNIQUE(task_id, ord)` implicitly creates an index on `(task_id, ord).` We only ever INSERT on `(task_id, ord)` or SELECT `task_id` and aggregate over `ord`.

I have some query plans from task upload counters https://github.com/divviup/janus/pull/2508#discussion_r1462320434.

Reply:
`UNIQUE` constraints generate a (unique) index on the relevant fields, so we have an index on `(task_id, ord)`.

See https://www.postgresql.org/docs/current/ddl-constraints.html#DDL-CONSTRAINTS-UNIQUE-CONSTRAINTS:

> Adding a unique constraint will automatically create a unique B-tree index on the column or group of columns listed in the constraint.